client/components/domains/register-domain-step/index.jsx

```diff
@@ -786,7 +791,13 @@ class RegisterDomainStep extends React.Component {
 						TRANSFERRABLE_PREMIUM,
 						UNKNOWN,
 					} = domainAvailability;
-					const isDomainAvailable = includes( [ AVAILABLE, UNKNOWN ], status );
+					// TODO: This may need to be cleaned up, as manually typed in premium domains for other TLDs
```

Comment:
We could do something like was done for ccTLDs at some point:
https://github.com/Automattic/wp-calypso/blob/74c6c2c5e51ae84caf50d1eef4f68708bae55039/config/wpcalypso.json#L36

So that we enable the TLDs individually... but then we'll want to enable all Donuts TLDs and suddenly have to add 200 lines to our config files 😆 I guess it'd be best to control this from the backend? The frontend could say "please check the availability of example.com - and, yes, I support premium domains", but the backend could do its own thing and then check TLD spec and be like "example.com is a premium domain name, BUT! we don't support premiums for .com yet, so I'll just mark this one as unavailable or not registrable premium".

Reply:
I was thinking about this more and think the issue is that pushing this into the existing availability API feels wrong, especially because the API is already telling us that the domain is `AVAILABLE_PREMIUM` -- it then depends on client context as to whether the domain is then purchasable.

I then wondered whether we could piggy-back on the existing `suggestions/tlds` API, but that only returns a flat list/array of TLDs, so doesn't have space to stash more complex data. But in thinking about this more, I think this might deserve its own API _and_ its own client-side state, as we're unlikely to need frequent updates to this data, and this is something that we are already tracking on the server. Does that make sense?